pkg/internal/tests/cross-tests/tf_driver.go

```diff
@@ -48,17 +48,25 @@ func newTFResDriver(t T, dir, providerName, resName string, res *schema.Resource
 	}
 }
 
-func (d *TfResDriver) coalesce(t T, x any) *tftypes.Value {
-	if x == nil {
-		return nil
-	}
-	objectType := convert.InferObjectType(sdkv2.NewSchemaMap(d.res.Schema), nil)
-	for k := range objectType.AttributeTypes {
-		objectType.OptionalAttributes[k] = struct{}{}
+func coalesceInputs(t T, schema map[string]*schema.Schema, config any) cty.Value {
```

Comment:
This was pre-existent to help construct cty.Value things but is ever-so-slightly sketchy. We need this permanently or temporarily? Why is `config any`? Is this for convenience or it's necessary? If we put some labor in here, could we rewrite all tests to use cty.Value to represent TF-side inputs so these other forms would not need to be supported?

Reply:
We need this temporarily. If we re-write all tests in `cross-tests` to be written in terms of `cty.Value`, then we can rip this out.

I didn't want to re-write all of our existing tests while iterating on this PR.

Reply:
Perfect. Comment on this in-source would be swell.